packages/yarnpkg-core/sources/Configuration.ts

```diff
@@ -1014,9 +1015,15 @@ export class Configuration {
     const configuration = new Configuration(startingCwd);
     configuration.importSettings(pickCoreFields(coreDefinitions));
 
+    const envList = rcFiles.map(rcFile => {
+      return [rcFile.cwd, configuration.getPackageEnv(rcFile.cwd)] as const;
```

Comment:
Should be async

Reply:
https://github.com/yarnpkg/berry/pull/4835/commits/d413eff313d84f2e8b453f40a5f36296876ff08b